rules.d/99-systemd.rules.in

```diff
@@ -22,6 +22,10 @@ SUBSYSTEM=="ubi", TAG+="systemd"
 
 SUBSYSTEM=="block", TAG+="systemd"
 
+# Make dm devices can replace existing symlinks with the same priority.
+# Otherwise, activating a device while another is being deactivated may fail.
+SUBSYSTEM=="block", KERNEL=="dm-*", ENV{.UDEV_REPLACE_SYMLINK_WITH_SAME_PRIORITY}="1"
```

Comment:
Can someone explain this in more detail? Is there a known failure scenario (except #28141)?

To my understanding "activating a device while another is being deactivated" means that 2 uevents are received simultaneously, where one is trying unregister a certain symlink and the other one is adding it. Why would that go wrong without `.UDEV_REPLACE_SYMLINK_WITH_SAME_PRIORITY`? If the "adding" event is processed first, it would register itself as a candidate for the symlink, and the "removing" event would remove itself and replace the symlink by the waiting candidate. Which is how this is supposed to work. There is a time window during which the symlink points to a device which is being removed; thus an operation that dereferences the symlink could grab the wrong device. But this isn't atomic anyway, and never has been.

IMO we can't reliably support a scenario like this. If we could, we should also be able to reliably support a scenario where several devices are activated simultaneously while one is deactivated. But we can't.

Well, I guess we could, but only if we added some additional priority criterion, like "the lower major/minor number wins".